File: Android App/app/src/main/java/revert/shredsheets/models/themes/DefaultTheme.java
package revert.shredsheets.models.themes;

import android.graphics.Color;
import android.graphics.Paint;

import revert.shredsheets.models.SessionModel;

public class DefaultTheme extends ShredsheetsTheme {

    private Boolean[] highlightedDegrees = null;

    private int[] degreeColors = null;

    private int borderColor = Color.parseColor("#222222");
    private float borderStrokeWidth = 2f;

    public int runnerBorderColor = Color.parseColor("#AAAABB");
    public float runnerStrokeWidth = 3.5f;
    private int[] intervalColors;


    public DefaultTheme() {
    }

    @Override
    public int getRunnerBorderColor() {
        return runnerBorderColor;
    }

    @Override
    public Boolean[] getDegreeHighlightingVector() {
        return highlightedDegrees != null ? highlightedDegrees :
                (highlightedDegrees = defaultVector); //supports up to max 12 degrees
    }

    @Override
    public void setDegreeHighlightingVector(Boolean[] values) {
        highlightedDegrees = values;
        SessionModel.getInstance().invalidateViews();
    }


    private Paint borderPaint = null;
    @Override
    public Paint getBorderPaint() {
        if (borderPaint == null) {
            borderPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
            borderPaint.setStyle(Paint.Style.STROKE);
            borderPaint.setColor(getBorderColor());
        }
        return borderPaint;
    }

    private Paint runnerBorderPaint = null;
    @Override
    public Paint getRunnerBorderPaint() {
        if (runnerBorderPaint == null) {
            runnerBorderPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
            runnerBorderPaint.setStyle(Paint.Style.STROKE);
            runnerBorderPaint.setColor(Color.parseColor("#111111"));
        }
        return runnerBorderPaint;
    }

    @Override
    public int getBorderColor() {
        return borderColor;
    }
    public void setBorderColor(int color) {
        borderColor = color;
    }

    private Paint scaleBorderPaint = null;
    @Override
    public Paint getScaleBorderPaint() {
        if (scaleBorderPaint != null) return scaleBorderPaint;

        scaleBorderPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        scaleBorderPaint.setColor(Color.rgb(200,200,200));
        scaleBorderPaint.setStyle(Paint.Style.STROKE);
        scaleBorderPaint.setStrokeWidth(10);
        return scaleBorderPaint;
    }

    public void setScaleBorderPaint(Paint scaleBorderPaint) {
        this.scaleBorderPaint = scaleBorderPaint;
    }

    @Override
    public int[] getDegreeColors() {
        return degreeColors != null ? degreeColors :
                (degreeColors = new int[]{
                        Color.rgb(55, 136, 203), //3788CB
                        Color.rgb(228, 210, 242),
                        Color.rgb(244, 218, 62), //F4DA3E
                        Color.rgb(181, 103, 245), //6AD81C
                        Color.rgb(106, 216, 28),
                        Color.rgb(100, 44, 118), //642C76
                        Color.rgb(255, 102, 255),
                        Color.rgb(175, 225, 170),
                        Color.rgb(100, 225, 90),
                        Color.rgb(30, 225, 10),
                        Color.rgb(225, 100, 10),
                        Color.rgb(225, 100, 75),
                });
    }

    public void setDegreeColors(int[] colors) {
        degreeColors = colors;
        SessionModel.getInstance().invalidateViews();
    }

    @Override
    public int getBaseFretColor(int fretNumber) {
        switch (fretNumber) {
            case 0:
            case 12:
            case 24:
                return Color.parseColor("#999999");
            case 3:
            case 5:
            case 7:
            case 9:
            case 15:
            case 17:
            case 19:
            case 21:
                return Color.parseColor("#CCCCDD");
            default:
                return Color.parseColor("#e9e9e9");
        }
    }

    @Override
    public int getFretboardRunnerPaintColor(int fretNumber) {
        switch (fretNumber) {
            case 0:
            case 12:
            case 24:
                return Color.parseColor("#404040");
            case 3:
            case 5:
            case 7:
            case 9:
            case 15:
            case 17:
            case 19:
            case 21:
                return Color.parseColor("#5c5c5c");
            default:
                return Color.parseColor("#d9d9d9");
        }
    }

    //Got replaced by a maximal color difference algorithm... may come back
    @Override
    public int getFretboardRunnerTextColor(int fretNumber) {
        switch (fretNumber) {
            case 0:
            case 12:
            case 24:
                return Color.parseColor("#333333");
            case 3:
            case 5:
            case 7:
            case 9:
            case 15:
            case 17:
            case 19:
            case 21:
                return Color.parseColor("#555555");
            default:
                return Color.parseColor("#777777");
        }
    }

    @Override
    public float getBorderStrokeWidth() {
        return borderStrokeWidth;
    }

    public void setBorderStrokeWidth(float borderStrokeWidth) {
        this.borderStrokeWidth = borderStrokeWidth;
    }

    @Override
    public int[] getIntervalColors() {
        if (intervalColors == null) intervalColors = new int[] {
                Color.parseColor("#777777"),
                Color.parseColor("#A0A0A0"),
                Color.parseColor("#BBBBBB"),
                Color.parseColor("#BBBBBB"),
                Color.parseColor("#BBBBBB") ,
                Color.parseColor("#BBBBBB") ,
                Color.parseColor("#BBBBBB")
        };
        return intervalColors;
    }

    public void setIntervalColors(int[] intervalColors) {
        this.intervalColors = intervalColors;
    }

    @Override
    public int getEmptyFretColor() {
        return Color.parseColor("#1e1918");
    }

    @Override
    public int getFretboardBorderColor() { return Color.parseColor("#030303");
    }

    @Override
    public int getTextStrokeColor() {
        return Color.parseColor("#090909");
    }

}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/themes/DefaultTheme.kt
package com.bm.shredsheets.models.themes

import com.badlogic.gdx.graphics.Color
import com.bm.shredsheets.models.Paint
import com.bm.shredsheets.models.SessionModel


class DefaultTheme() : ShredsheetsTheme() {
    private var highlightedDegrees: BooleanArray? = null

    //80A8C9
    //3788CB
    //6AD81C
    //2943C6
    //642C76
    override var degreeColors: Array<Color> = arrayOf(
            Color(55f / 255f, 136f / 255f, 203f / 255f, 1f),  //80A8C9
            Color(211f / 255f, 181f / 255f, 233f / 255f, 1f),
            Color(244f / 255f, 218f / 255f, 62f / 255f, 1f),  //3788CB
            Color(181f / 255f, 103f / 255f, 245f / 255f, 1f),  //6AD81C
            Color(117f / 255f, 230f / 255f, 4f / 255f, 1f),  //2943C6
            Color(100f / 255f, 44f / 255f, 118f / 255f, 1f),  //642C76
            Color(255f / 255f, 102f / 255f, 255f / 255f, 1f),
            Color(175f / 255f, 225f / 255f, 170f / 255f, 1f),
            Color(100f / 255f, 225f / 255f, 90f / 255f, 1f),
            Color(30f / 255f, 225f / 255f, 10f / 255f, 1f),
            Color(225f / 255f, 100f / 255f, 10f / 255f, 1f),
            Color(225f / 255f, 100f / 255f, 75f / 255f, 1f))

    override var borderColor: Color = Color(.75f, .75f, .75f, 1f)
    override var borderStrokeWidth: Float = 2f
    override var runnerBorderColor: Color = Color(.65f, .65f, .75f, 1f)
    var runnerStrokeWidth: Float = 3.5f

    override var intervalColors: Array<Color> = arrayOf(
                Color(.35f, .35f, .35f, 1f),
                Color(.55f, .55f, .55f, 1f),
                Color(.75f, .75f, .75f, 1f),
                Color(.75f, .75f, .75f, 1f),
                Color(.75f, .75f, .75f, 1f),
                Color(.75f, .75f, .75f, 1f),
                Color(.75f, .75f, .75f, 1f))

    //supports up to max 12 degrees
    override var degreeHighlightingVector: BooleanArray?
        get() {
            return if (highlightedDegrees != null) highlightedDegrees else (defaultVector.also {
                highlightedDegrees = it
            }) //supports up to max 12 degrees
        }
        set(values) {
            highlightedDegrees = values
            SessionModel.instance.invalidateViews()
        }

    override var borderPaint: Paint? = null
        get() {
            if (field == null) {
                field = Paint()
                field!!.color = borderColor
            }
            return field
        }

    override var runnerBorderPaint: Paint? = null
        get() {
            if (field == null) {
                field = Paint()
                field!!.color = runnerBorderColor
            }
            return field
        }


    override var scaleBorderPaint: Paint? = null
        get() {
            if (field == null) {
                field = Paint()
                field!!.color = Color(.78f, .78f, .78f, 1f)
                field!!.strokeWidth = 10f
            }
            return scaleBorderPaint
        }


    override fun getBaseFretColor(fretNumber: Int): Color {
        when (fretNumber) {
            0, 12, 24 -> return  Color(.61f, .61f, .61f, 1f)
            3, 5, 7, 9, 15, 17, 19, 21 -> return Color(.81f, .81f, .88f, 1f)
            else -> return Color(.93f, .93f, .93f, 1f)
        }
    }

    override fun getFretboardRunnerPaintColor(fretNumber: Int): Color {
        when (fretNumber) {
            0, 12, 24 -> return Color(.25f, .25f, .25f, 1f)
            3, 5, 7, 9, 15, 17, 19, 21 -> return Color(.36f, .36f, .36f, 1f)
            else -> return Color(.93f, .93f, .93f, 1f)
        }
    }

    //Got replaced by a maximal color difference algorithm... may come back
    override fun getFretboardRunnerTextColor(fretNumber: Int): Color {
        when (fretNumber) {
            0, 12, 24 -> return Color(.2f, .2f, .2f, 1f)
            3, 5, 7, 9, 15, 17, 19, 21 -> return Color(.34f, .34f, .34f, 1f)
            else -> return Color(.5f, .5f, .5f, 1f)
        }
    }

    override val emptyFretColor: Color
        get() {
            return Color(.10f, .10f, .10f, 1f)
        }
    override val fretboardBorderColor: Color
        get() {
            return Color(.02f, .02f, .02f, 1f)
        }
    override val textStrokeColor: Color
        get() {
            return Color(.05f, .05f, .05f, 1f)
        }
}
